components/Themes/src/MDCTonalColorScheme.m

```diff
@@ -0,0 +1,56 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCTonalColorScheme.h"
+
+@implementation MDCTonalColorScheme
+
+- (nonnull instancetype)initWithPrimaryTonalPalette:(nonnull MDCTonalPalette *)primaryTonalPalette
+                              secondaryTonalPalette:(nonnull MDCTonalPalette *)secondaryTonalPalette
+    {
+  self = [super init];
+  if (self) {
+    _primaryTonalPalette = primaryTonalPalette;
+    _secondaryTonalPalette = secondaryTonalPalette;
+  }
+  return self;
+}
+
+- (UIColor *)primaryColor {
+  return _primaryTonalPalette.colors[_primaryTonalPalette.mainColorIndex];
```

Comment:
Now that we've got the direct UIColor accessors, consider:
return self.primaryTonalPalette.mainColor;
Here and below.